packages/labs/context/src/lib/controllers/context-consumer.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ContextRequestEvent} from '../context-request-event.js';
+import {ContextKey, ContextType} from '../context-key.js';
+import {ReactiveController, ReactiveElement} from 'lit';
+
+export type DispatchOptions = 'connected' | 'updated' | 'raf' | 'microtask';
+
+/**
+ * ContextConsumer is a ReactiveController which binds a custom-element's
+ * lifecycle to the Context API. When an element is connected to the DOM it
+ * will emit the context-request event, invoking the callback set on the
+ * controller when the context request is satisfied. It will also call
+ * the dispose method provided by the Context API when the element is
+ * disconnected.
+ */
+export class ContextConsumer<
+  Context extends ContextKey<unknown, unknown>,
+  HostElement extends ReactiveElement
+> implements ReactiveController
+{
+  private provided = false;
+
+  constructor(
+    protected host: HostElement,
+    private context: Context,
+    private callback: (
```

Comment:
[Question] This signature seem to be used at a few places. Maybe it should have a name? Unless this inline type is for the typings rollup

Reply:
If you mean the callback signature, maybe, its now only used in 2 places one of which is readonly, so am leaving this as is for now.